ESP8266/ESP8266.cpp

```diff
@@ -294,27 +299,29 @@ void ESP8266::setTimeout(uint32_t timeout_ms)
     _parser.setTimeout(timeout_ms);
 }
 
-bool ESP8266::readable()
-{
-    return _serial.readable();
-}
-
-bool ESP8266::writeable()
-{
-    return _serial.writeable();
-}
-
 void ESP8266::attach(Callback<void()> func)
 {
-    _serial.attach(func);
+    _serial.sigio(func);
 }
 
 bool ESP8266::recv_ap(nsapi_wifi_ap_t *ap)
 {
-    int sec;
-    bool ret = _parser.recv("+CWLAP:(%d,\"%32[^\"]\",%hhd,\"%hhx:%hhx:%hhx:%hhx:%hhx:%hhx\",%d", &sec, ap->ssid,
-                            &ap->rssi, &ap->bssid[0], &ap->bssid[1], &ap->bssid[2], &ap->bssid[3], &ap->bssid[4],
-                            &ap->bssid[5], &ap->channel);
+    // Newlib-nano doesn't support %hhx or SCNu8 scan formats,
```

Comment:
%hhx missing from newlib nano is concerning and I'm not really sold on supporting newlib nano for this driver since we rely on the stdlib's formating so much. But we can move forward with this if you want.